include/worker.h

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright © 2017 Drew Walters <drewwalters96@gmail.com>
+ * SPDX-License-Identifier: (BSD-3-Clause OR GPL-2.0)
+ */
+
+#ifndef WORKER_H
+#define WORKER_H
+
+/* Queue worker function capacity. */
+#define MAX_WORKERS 1
+
+/*
+ * Type definition of worker function added to queue by interrupt handler.
+ */
+typedef int (*worker)(void *);
+
+/*
+ * Queues a worker function when called by an interrupt handler.
+ */
+void queue_worker(worker);
+
+/*
+ * Executes each worker function in the queue and panics if a worker function
+ * returns an error status.
+ */
+void process_workers(void);
+
+/*
+ * Queue of worker functions.
+ */
+static int (*workers[MAX_WORKERS])(void *);
```

Comment:
This needs to be in the C source file; otherwise there will be a copy of the array allocated in every file that includes the header.

Reply:
Fixed